### PR TITLE
[FT-AMS-102020] Emma Northridge

### PR DESCRIPTION
**Pull request by emmanorthridge**
Logic part done

### CODE REVIEW

File: js/memory.js [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
-}
\ No newline at end of file
+  shuffleCards() {
+    let j, x, i;
```
Comment (wax-bear) [inline js/memory.js:9]: Could the variables "j" and "x" be named more expressively? What do these variables represent, or contain?

File: js/memory.js [hunk 2]
```diff
@@ -1,9 +1,40 @@
...
+    }
+  }
+
+  checkIfPair(card1, card2) {
+    if (card1 === card2) {
+      this.pairsGuessed += 1;
+      this.pairsClicked += 1;
+      return true;
+    } else {
+      this.pairsClicked += 1;
```
Comment (wax-bear) [inline js/memory.js:28]: Line 25 and 28 contain the same code. This violates the Don't Repeat Yourself principle (DRY). Perhaps delete both and place this code on the first line of the function body.

File: js/memory.js [hunk 3]
```diff
@@ -1,9 +1,40 @@
...
+      return false;
+    }
+  }
+
+  isFinished() {
```
Comment (wax-bear) [inline js/memory.js:33]: Is isFinished called somewhere?

File: js/memory.js [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+    if (this.pairsGuessed === this.cards.length / 2) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```
Comment (wax-bear) [inline js/memory.js:39]: Good stuff Emma!
index.js is not altered yet, that explains why the UI is not working as expected.
Please let me know if you have any question.

Comment (emmanorthridge) [reply js/memory.js:39]: Thanks!
I haven't started the index.js file yet. 
I'll work on that this morning :)

**Comment by emmanorthridge** [comment]
@wax-bear I have now completed the iteration :)

**Pull request closed without merging**